### PR TITLE
Bump Serving manifests to 1.14

### PR DESCRIPTION
**Pull request by skonto**
Fixes JIRA #

## Proposed Changes

- Similar to https://github.com/openshift-knative/serverless-operator/pull/2524
- On certain OS like Fedora 40 wget2 is used now, see https://github.com/coreos/fedora-coreos-tracker/issues/1626, which breaks current setup with relative paths. I get for example:

> wget  --no-check-certificate https://raw.githubusercontent.com/openshift-knative/eventing-kafka-broker/release-v1.12/openshift/release/artifacts/eventing-kafka-controller.yaml -O ./knative-operator/hack/../../knative-operator/deploy/resources/knativekafka/controller/eventing-kafka-controller.yaml
> WARNING: The certificate is NOT trusted. The certificate issuer is unknown. 
> Internal error: Unexpected relative path: './knative-operator/hack/..'
So this patch uses real paths.
- run `make generated-files`

### CODE REVIEW

File: olm-catalog/serverless-operator/project.yaml [hunk 1]
```diff
@@ -31,14 +31,14 @@ requirements:
         max: '4.15'
         label: 'v4.12'
 dependencies:
-    serving: knative-v1.12
+    serving: knative-v1.14
```
Comment (skonto) [inline olm-catalog/serverless-operator/project.yaml:34]: Do we need a mechanism to ignore this gap 1.12 -> 1.14 anywhere? What is the status with version checking?

Comment (ReToCode) [reply olm-catalog/serverless-operator/project.yaml:34]: Should be fine (not tested). We removed the semver check in S-O.

File: openshift-knative-operator/cmd/operator/kodata/ingress/1.14/kourier/1-net-kourier.yaml [hunk 2]
```diff
@@ -97,15 +97,6 @@ data:
                                       exact: GET
                               route:
                                 cluster: service_stats
-                            - match:
```
Comment (ReToCode) [inline openshift-knative-operator/cmd/operator/kodata/ingress/1.14/kourier/1-net-kourier.yaml:100]: @skonto what is the status of your kourier graceful shutdown testing?

**Comment by ReToCode** [comment]
LGTM.

File: openshift-knative-operator/cmd/operator/kodata/ingress/1.14/kourier/1-net-kourier.yaml (hunk 2 above)
Comment (skonto) [reply openshift-knative-operator/cmd/operator/kodata/ingress/1.14/kourier/1-net-kourier.yaml:100]: I need to do a PR against net-kourier https://github.com/openshift-knative/net-kourier/pull/130#issuecomment-2084532432 cherry-pick failed.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**